### PR TITLE
xds: optimize fault injection filter with empty config

### PR DESCRIPTION
**Pull request by dfawley**

### CODE REVIEW

File: xds/internal/httpfilter/fault/fault.go [hunk 1]
```diff
@@ -125,7 +125,12 @@ func (builder) BuildClientInterceptor(cfg, override httpfilter.FilterConfig) (ir
 		}
 	}
 
-	return &interceptor{config: c.config}, nil
+	icfg := c.config
+	if (icfg.GetMaxActiveFaults() != nil && icfg.GetMaxActiveFaults().GetValue() == 0) ||
```
Comment (easwars) [inline xds/internal/httpfilter/fault/fault.go:129]: `GetValue()` does handle a `nil` receiver.
```golang
func (x *UInt32Value) GetValue() uint32 {
	if x != nil {
		return x.Value
	}
	return 0
}
```
Do, we still need the `icfg.GetMaxActiveFaults() != nil` check?

Comment (dfawley) [reply xds/internal/httpfilter/fault/fault.go:129]: Yes.  If `GetMaxActiveFaults` is _explicitly_ set to zero, it is disabled.  But if it is unset, there is no limit.

**Comment by dfawley** [comment]
Re-assigning to myself; I need to add tests.

**Comment by dfawley** [comment]
> Re-assigning to myself; I need to add tests.

Done!

**Review by easwars — APPROVED** [review]

**Comment by dfawley** [comment]
(force push for flaky test fixes which flaked twice so far)

**Pull request merged**